torch/csrc/jit/interpreter_autograd_function.h

```diff
@@ -0,0 +1,37 @@
+#pragma once
+
+#include "torch/csrc/jit/interpreter.h"
+#include "torch/csrc/autograd/function.h"
+#include "torch/csrc/autograd/functions/utils.h"
+#include "torch/csrc/autograd/functions/basic_ops.h"
+namespace torch { namespace jit {
+struct InterpreterAutogradFunction : public autograd::Function {
+  InterpreterAutogradFunction(const jit::Function & function)
+  : interp_(function) {}
+  InterpreterAutogradFunction(const Interpreter & interp_, autograd::FunctionFlags && f)
+  : autograd::Function(std::move(f)), interp_(interp_) {}
+
+  virtual void willReleaseVariables() override {
+    keep_graph = false;
+  }
+  virtual autograd::variable_list apply(const autograd::variable_list& inputs) override {
```

Comment:
I a not confident that Variable flags are being propagated correctly as the interpreter runs. @apaszke can you let me know what I am suppose to be doing with them here. I imagine we should be just saving the flags from the trace and reproducing them here.

Reply:
Actually, assuming that you've accurately got all the inputs, I think this actually will DTRT.

Reply:
No, this won't work. We have the [information necessary to reproduce the flags in `TracingState`](https://github.com/pytorch/pytorch/blob/master/torch/csrc/jit/tracer_state.h#L68), so it's possible to do right now. I'm fine with merging as is and I can fix that later.